kubernetes/cmsweb/monitoring/logstash.conf

```diff
@@ -78,6 +78,26 @@ filter {
   }
   if "crabserver" in [tags] {
       mutate { replace => { "type" => "crabserver" } }
+      grok {
+        match => { "message" => '\[%{NOTSPACE:tstamp}\] %{DATA:backend} %{IPORHOST:clientip} "%{WORD:method} %{NOTSPACE:request} %{DATA:httpversion}" %{NUMBER:code:int}[^\[]* \[data: %{NUMBER:bytes_sent:int} in %{NUMBER:bytes_received:int} out %{NUMBER:time_spent_ms:int} us \] \[auth: %{DATA} "%{DATA:dn}".*\] \[ref: "%{DATA}.*" "%{DATA:client}" \]' }
+      }
+      grok {
+        match => { "request" => '/%{WORD:system}%{UNIXPATH:uri_path}%{URIPARAM:uri_params}?' }
+      }
+      if ![uri_params] {
+          ruby { code => "event.set('uri_params','')" }
+      }
+      grok { match => { "uri_path" => '/.*/%{DATA:api}$' } }
+      date {
+         match => [ "tstamp", "dd/MMM/yyyy:HH:mm:ss" ]
+         target => "date_object"
+      }
+      ruby {
+         code => "event.set('rec_timestamp',event.get('date_object').to_i)
+                  event.set('rec_date',event.get('date_object'))
+                 "
+      }
```

Comment:
@novicecpp 
There is a  "timestamp" keyword of MONIT (like producer, type). They set `metadata.timestamp` as `timestamp` value if we set it. `metadata.timestamp` is the default date time of ElasticSearch index in Kibana. If you did not set it as you did, MONIT will set their own timestamp which will be the time when our data is reached to MONIT.

So, you can set timestamp with the event time like we did in "frontend" tag. You can use:
```
ruby {
         code => "event.set('timestamp',(event.get('date_object').to_f * 1000).to_i)
                  event.set('rec_date',event.get('date_object'))
                 "
      }

# And, add timestamp to message part in the output:
 message => '{"producer": "cmswebk8s", "type": "crabserver", ... , "timestamp": "%{timestamp}"}'

# Remove "rec_timestamp":"%{rec_timestamp}",
# rec_timestamp is not necessary in your case
```

Date fields will be like in ES:
`data.@timestamp`: Logstash timestamp
`metadata.kafka_timestamp`: MONIT kafka timestamp
`metadata.timestamp`: Event/Log time which we parsed from log